lib/src/parsers/example-parser.spec.ts

```diff
@@ -0,0 +1,99 @@
+import { extractJSDocExamples } from "./example-parser";
+import { createProjectFromExistingSourceFile } from "../spec-helpers/helper";
+import {
+  getJsDoc,
+  parseTypeReferencePropertySignaturesOrThrow
+} from "./parser-helpers";
+import { JSDoc, TypeAliasDeclaration } from "ts-morph";
+import { StringType, Type, TypeKind } from "../types";
+
+type JSDocNodeMapping = {
+  ParsedExamples: JSDoc[];
+  MismatchedExampleAndIntegerType: JSDoc[];
+  MismatchedExampleAndStringWithQuotesType: JSDoc[];
+  [typeName: string]: JSDoc[];
+};
```

Comment:
I don't see much of an advantage in holding on to an object of this type. Each `test` can just deal with one `property` of the `type`. See related comment from `__spec-examples__/examples.ts`